fiftyone/server/events.py

```diff
@@ -256,18 +256,53 @@ async def _initialize_listener(payload: ListenPayload) -> InitializedListener:
         _app_count += 1
 
     current = state.dataset.name if state.dataset else None
-    if is_app and payload.initializer != current:
-        if payload.initializer:
+    current_saved_view = state.view_name
+    if not isinstance(payload.initializer, fos.StateDescription):
+        update = False
+        if (
+            payload.initializer.dataset
+            and payload.initializer.dataset != current
+        ):
+            update = True
             try:
-                state.dataset = fo.load_dataset(payload.initializer)
+                state.dataset = fo.load_dataset(payload.initializer.dataset)
+                state.selected = []
+                state.selected_labels = []
+                state.view = None
+                state.view_name = None
             except:
                 state.dataset = None
-            state.selected = []
-            state.selected_labels = []
-            state.view = None
-            state.saved_view_slug = None
-            state.changing_saved_view = False
+                state.selected = []
+                state.selected_labels = []
+                state.view = None
+                state.view_name = None
+            else:
+                if payload.initializer.view:
+                    try:
+                        state.view = state.dataset.load_saved_view(
+                            payload.initializer.view
+                        )
+                        state.selected = []
+                        state.selected_labels = []
+                        state.view_name = payload.initializer.view
+                    except:
+                        pass
+        elif (
+            payload.initializer.view
+            and payload.initializer.view != current_saved_view
```

Comment:
if `payload.initializer.view` is from the url, its value is`view.slug`. Is `state.view_name` always going to be the slug? or only set here? If so, maybe rename it to slug to prevent setting it elsewhere in the code using `view.name`

Reply:
It will come from the URL but will have already been decoded via [decodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent). The server only needs to know the name.